mobile-app/app/screens/AppNavigator/screens/Portfolio/components/CreateOrEditAddressLabelFormV2.tsx

```diff
@@ -0,0 +1,298 @@
+    if (input !== undefined) {
+      if (input.trim().length > 40) {
+        setLabelInputErrorMessage('Invalid label. Maximum of 40 characters.')
+        return false
+      }
+
+      if (isAddressBook && input.trim() === '') {
+        setLabelInputErrorMessage('Please enter an address label')
+        return false
+      }
+    }
+
+    setLabelInputErrorMessage('')
+    return true
+  }
+
+  const validateAddressInput = (input: string): boolean => {
+    const decodedAddress = fromAddress(input, networkName)
+    if (decodedAddress === undefined) {
+      setAddressInputErrorMessage('Please enter a valid address')
+      return false
+    }
+    if (addressBook?.[input.trim()] !== undefined) {
+      setAddressInputErrorMessage('This address already exists in your address book, please enter a different address')
+      return false
+    }
+    setAddressInputErrorMessage('')
+    return true
+  }
+
+  const handleSubmit = async (): Promise<void> => {
+    if (!isAddressBook) {
+      handleEditSubmit()
+    } else {
+      handleCreateSubmit()
+    }
+  }
+
+  const handleEditSubmit = (): void => {
+    if (labelInput === undefined ||
+      address === undefined ||
+      !validateLabelInput(labelInput)) {
+      return
+    }
+    onSaveButtonPress({
+      [address]: {
+        address,
+        label: labelInput.trim(),
+        isMine: true
+      }
+    })
+  }
+
+  // Passcode prompt on create
+  const dispatch = useAppDispatch()
+  const { data: { type: encryptionType } } = useWalletNodeContext()
+  const isEncrypted = encryptionType === 'MNEMONIC_ENCRYPTED'
+  const logger = useLogger()
+  const handleCreateSubmit = useCallback(() => {
+    if (!isEncrypted ||
+      addressInput === undefined ||
+      labelInput === undefined ||
+      !validateLabelInput(labelInput) ||
+      !validateAddressInput(addressInput)) {
+      return
+    }
+
+    const auth: Authentication<string[]> = {
+      consume: async passphrase => await MnemonicStorage.get(passphrase),
+      onAuthenticated: async () => {
+        onSaveButtonPress({
+          [addressInput]: {
+            address: addressInput,
+            label: labelInput.trim(),
+            isMine: false
+          }
+        })
+      },
+      onError: e => logger.error(e),
+      message: translate('screens/Settings', 'Enter passcode to continue'),
+      loading: translate('screens/Settings', 'Verifying access')
+    }
+    dispatch(authentication.actions.prompt(auth))
+  }, [dispatch, isEncrypted, addressInput, labelInput, onSaveButtonPress])
+
+  const isSaveDisabled = (): boolean => {
+    if (isAddressBook && (addressInput === undefined || labelInput === undefined || labelInputErrorMessage !== '' || addressInputErrorMessage !== '')) {
+      return true
+    }
+    if (!isAddressBook && (labelInput === undefined || labelInput === addressLabel?.label || labelInputErrorMessage !== '')) {
+      return true
+    }
+
+    return false
+  }
+
+  return (
+    <ScrollView
```

Comment:
Should we use v2?

Reply:
Already using v2 
![image](https://user-images.githubusercontent.com/89639796/180167467-d8a5823c-4485-4ca0-bef0-80067c8357a4.png)